devices/HumanDataCollector/HumanDataCollector.cpp

```diff
@@ -0,0 +1,663 @@
+            yError() << "The viewed IHumanState interface from " << deviceName << " polydriver device might not be ready";
+            return false;
+        }
+
+        yInfo() << LogPrefix << deviceName << "attach() successful";
+        pImpl->isStateProviderDevice = true;
+    }
+
+    if (deviceName == "human_wrench_provider") {
+
+        // Attach to IHumanWrench interface from HumanWrenchProvider
+        if (pImpl->iHumanWrenchMeasurements || !poly->view(pImpl->iHumanWrenchMeasurements) || !pImpl->iHumanWrenchMeasurements) {
+            yError() << LogPrefix << "Failed to view IHumanWrench interface from the " << deviceName << " polydriver device";
+            return false;
+        }
+
+        // Check the viewed interface
+        if (pImpl->iHumanWrenchMeasurements->getNumberOfWrenchSources() != pImpl->iHumanWrenchMeasurements->getWrenchSourceNames().size()) {
+            yError() << "The viewed IHumanWrench interface from " << deviceName << " polydriver device might not be ready";
+            return false;
+        }
+
+        yInfo() << LogPrefix << deviceName << "attach() successful";
+        pImpl->isWrenchProviderDevice = true;
+
+    }
+
+    if (deviceName == "human_dynamics_estimator") {
+
+        // Attach to IHumanWrench interface from HumanDynamicsEstimator
+        if (pImpl->iHumanWrenchEstimates || !poly->view(pImpl->iHumanWrenchEstimates) || !pImpl->iHumanWrenchEstimates) {
+            yError() << LogPrefix << "Failed to view IHumanWrench interface from the " << deviceName << " polydriver device";
+            return false;
+        }
+
+        // Check the viewed interface
+        if (pImpl->iHumanWrenchEstimates->getNumberOfWrenchSources() != pImpl->iHumanWrenchEstimates->getWrenchSourceNames().size()) {
+            yError() << "The viewed IHumanWrench interface from " << deviceName << " polydriver device might not be ready";
+            return false;
+        }
+
+        // Attach to IHumanDynamics interface from HumanDynamicsEstimator
+        if (pImpl->iHumanDynamics || !poly->view(pImpl->iHumanDynamics) || !pImpl->iHumanDynamics) {
+            yError() << LogPrefix << "Failed to view IHumanDynamics interface from the  " << deviceName << " polydriver device";
+            return false;
+        }
+
+        // Check the viewed interface
+        if (pImpl->iHumanDynamics->getNumberOfJoints() == 0 || pImpl->iHumanDynamics->getNumberOfJoints() != pImpl->iHumanDynamics->getJointNames().size()) {
+            yError() << "The viewed IHumanDynamics interface from " << deviceName << " polydriver device might not be ready";
+            return false;
+        }
+
+        yInfo() << LogPrefix << deviceName << "attach() successful";
+        pImpl->isDynamicsEstimatorDevice = true;
+    }
+
+    return true;
+}
+
+bool HumanDataCollector::detach()
+{
+    while (isRunning()) {
+        stop();
+    }
```

Comment:
Just a curiosity, do we really need this check?

Reply:
I believe we settled on this check after investigating the termination process of devices https://github.com/robotology/human-dynamics-estimation/issues/109

Reply:
I suspected it was something related to termination problems but I was not sure. Thanks!